apps/aecore/src/aec_conductor.erl

```diff
@@ -287,9 +290,11 @@ handle_call(is_leader, _From, State = #state{ consensus = Cons }) ->
 handle_call(reinit_chain, _From, State1 = #state{ consensus = Cons }) ->
     %% NOTE: ONLY FOR TEST
     ok = reinit_chain_state(),
-    TopBlockHash = aec_chain:top_block_hash(),
+    TopBlock = aec_chain:top_block(),
+    {ok, TopBlockHash} = aec_blocks:hash_internal_representation(TopBlock),
```

Comment:
Is there no risk here? Now we re-compute the headers, before we read it from the database. 
Why the change? (just curious.)

Reply:
The top block is read from the db using the top block hash, so I suppose there is no risk here.
```
-spec top_block() -> 'undefined' | aec_blocks:block().
top_block() ->
    case top_block_hash() of
        undefined -> undefined;
        Hash -> aec_db:get_block(Hash)
    end.

-spec top_block_hash() -> 'undefined' | binary().
top_block_hash() ->
    aec_db:get_top_block_hash().
```